lucene/analysis/common/src/java/org/apache/lucene/analysis/hunspell/Dictionary.java

```diff
@@ -1085,8 +1062,9 @@ private void writeNormalizedWordEntry(
     int sep = flagSep < 0 ? morphSep : flagSep;
 
     CharSequence toWrite;
-    if (needsInputCleaning) {
-      cleanInput(line, sep, reuse);
+    String beforeSep = line.substring(0, sep);
+    if (needsInputCleaning(beforeSep)) {
```

Comment:
Removed the `cleanInput` overload with prefixLen as it complicates code and doesn't seem to be of benefit